sonic-net/sonic-buildimage
# [Netberg] Added Trident 3/4 based Netberg Aurora switches

#### Why I did it

Support for Broadcom Trident 3/4 based Netberg Aurora platforms:

Netberg Aurora 221:
ASIC: Trident 3 Broadcom BCM56275
Pors: 48x 1G BASE-T, 8x 10G SFP+ ,

Netberg Aurora 621:
ASIC: Broadcom Trident 3 Broadcom BCM56770
Ports: 48x 25GbE SFP28, 6x 100GbE QSFP28,

Netberg Aurora 721:
ASIC: Broadcom Trident 3 BCM56870
Ports:  32x 100GbE QSFP28 ports

Netberg Aurora 830:
ASIC:  Broadcom Trident 4 BCM56880
Ports: 32x 400G QSFPDD + 4x 10G SFP+

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Added specification to device/netberg directory with pddf description 
Added platform/broadcom/sonic-platform-modules-netberg contains scripts and sonic_platform packages.
Added platform/broadcom/platform-modules-netberg.mk to include switches related IDs
Modified rules.dep, rules.mk, one-image.mk in platform/broadcom/rules.dev

#### How to verify it
Build SONiC

Install the image on the device and verify the related components are installed and shown correctly.

Verification output:
[221-show.txt](https://github.com/sonic-net/sonic-buildimage/files/15087609/221-show.txt), 
[621-show-a-240723.txt](https://github.com/user-attachments/files/16495376/621-show-a-240723.txt), 
[830-show-a-240724.txt](https://github.com/user-attachments/files/16495378/830-show-a-240724.txt), 
[721-show-a-240723.txt](https://github.com/user-attachments/files/16495379/721-show-a-240723.txt)



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
